src/frontegg-mock/src/lib.rs

```diff
@@ -965,6 +1013,229 @@ async fn handle_delete_group_mapping(
+                    return Err(StatusCode::NOT_FOUND);
+                }
+            }
+        }
+        Ok(StatusCode::CREATED)
+    } else {
+        Err(StatusCode::NOT_FOUND)
+    }
+}
+
+// https://docs.frontegg.com/reference/groupscontrollerv1_removerolesfromgroup
+async fn handle_remove_roles_from_group(
+    State(context): State<Arc<Context>>,
+    Path(group_id): Path<String>,
+    Json(payload): Json<AddRolesToGroupParams>,
+) -> Result<StatusCode, StatusCode> {
+    let mut groups = context.groups.lock().unwrap();
+
+    if let Some(group) = groups.get_mut(&group_id) {
+        group
+            .roles
+            .retain(|role| !payload.role_ids.contains(&role.id));
+        Ok(StatusCode::OK)
+    } else {
+        Err(StatusCode::NOT_FOUND)
+    }
+}
+
+#[derive(Deserialize)]
+struct AddRolesToGroupParams {
+    #[serde(rename = "roleIds")]
+    role_ids: Vec<String>,
+}
+
+// https://docs.frontegg.com/reference/groupscontrollerv1_adduserstogroup
+async fn handle_add_users_to_group(
+    State(context): State<Arc<Context>>,
+    Path(group_id): Path<String>,
+    Json(payload): Json<AddUsersToGroupParams>,
+) -> Result<StatusCode, StatusCode> {
+    let mut groups = context.groups.lock().unwrap();
+    if let Some(group) = groups.get_mut(&group_id) {
+        for user_id in payload.user_ids {
+            if !group.users.iter().any(|u| u.id == user_id) {
+                group.users.push(ScimUser {
+                    id: user_id,
+                    name: "".to_string(),
+                    email: "".to_string(),
+                });
+            }
+        }
+        Ok(StatusCode::CREATED)
+    } else {
+        Err(StatusCode::NOT_FOUND)
+    }
+}
+
+// https://docs.frontegg.com/reference/groupscontrollerv1_removeusersfromgroup
+async fn handle_remove_users_from_group(
+    State(context): State<Arc<Context>>,
+    Path(group_id): Path<String>,
+    Json(payload): Json<RemoveUsersFromGroupParams>,
+) -> StatusCode {
+    let mut groups = context.groups.lock().unwrap();
+
+    if let Some(group) = groups.get_mut(&group_id) {
+        group
+            .users
+            .retain(|user| !payload.user_ids.contains(&user.id));
+        StatusCode::OK
+    } else {
+        StatusCode::NOT_FOUND
+    }
+}
+
+#[derive(Deserialize)]
+struct AddUsersToGroupParams {
+    #[serde(rename = "userIds")]
+    user_ids: Vec<String>,
+}
+
+#[derive(Deserialize)]
+struct RemoveUsersFromGroupParams {
+    #[serde(rename = "userIds")]
+    user_ids: Vec<String>,
+}
+
+async fn handle_list_scim_configurations(
+    State(context): State<Arc<Context>>,
+) -> Json<Vec<SCIM2Configuration>> {
+    let configs = context.scim_configurations.lock().unwrap();
+    Json(configs.values().cloned().collect())
+}
+
+async fn handle_create_scim_configuration(
+    State(context): State<Arc<Context>>,
+    Json(mut new_config): Json<SCIM2Configuration>,
+) -> Json<SCIM2Configuration> {
+    new_config.id = Uuid::new_v4().to_string();
```

Comment:
As always, it would be better to use separate types for requests and storage/response, especially when you're replacing a required value from the payload with a random value.